java/arcs/core/analysis/ClaimDeduction.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.expression.Expression
+
+internal typealias Path = List<AccessPath.Selector.Field>
+internal typealias Paths = List<Path>
+internal typealias ClaimDerivations = Map<Path, Set<Path>>
+
+/** A visitor that accumulates field [Paths] from a Paxel [Expression]. */
+class ExpressionPathAccumulator : Expression.Visitor<Paths> {
```

Comment:
I don't understand why we need 2 separate visitors and KDoc does not explain this. It seems to me we should be able to perform the test at hand with just one.